test/unittest/backends/mongodb/test_query.py

```diff
@@ -1,5 +1,111 @@
+import random
+from test.db_config import DBConfig
+
+import numpy as np
+import pytest
+
 from superduperdb.backends.mongodb import query as q
+from superduperdb.backends.mongodb.query import Collection
+from superduperdb.base.config import BytesEncoding
 from superduperdb.base.document import Document
+from superduperdb.components.schema import Schema
+from superduperdb.ext.numpy.encoder import array
+
+
+@pytest.fixture
+def schema(request):
+    bytes_encoding = request.param if hasattr(request, 'param') else None
+    if bytes_encoding is None:
+        return None
+
+    array_tensor = array("float64", shape=(32,), bytes_encoding=bytes_encoding)
+    schema = Schema(
+        identifier='documents',
+        fields={
+            "x": array_tensor,
+            "z": array_tensor,
+        },
+    )
+    return schema
+
+
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def test_mongo_without_schema(db):
+    collection_name = "documents"
+    array_tensor = array("float64", shape=(32,))
+    db.add(array_tensor)
+    data = []
+
+    for id_ in range(5):
+        x = np.random.rand(32)
+        y = int(random.random() > 0.5)
+        z = np.random.randn(32)
+        data.append(
+            Document(
+                {
+                    "id": id_,
+                    "x": array_tensor(x),
+                    "y": y,
+                    "z": array_tensor(z),
+                }
+            )
+        )
+
+    db.execute(
+        Collection(collection_name).insert_many(data),
+    )
+    collection = Collection(collection_name)
+    r = collection.find_one().execute(db)
+    rs = list(collection.find().execute(db))
+
+    rs = sorted(rs, key=lambda x: x['id'])
+
+    assert np.array_equal(r['x'].x, data[0]['x'].x)
+    assert np.array_equal(r['z'].x, data[0]['z'].x)
+
+    assert np.array_equal(rs[0]['x'].x, data[0]['x'].x)
+    assert np.array_equal(rs[0]['z'].x, data[0]['z'].x)
+
+
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+@pytest.mark.parametrize(
+    "schema", [BytesEncoding.BASE64, BytesEncoding.BYTES], indirect=True
+)
+def test_mongo_schema(db, schema):
+    collection_name = "documents"
+    data = []
+
+    for id_ in range(5):
+        x = np.random.rand(32)
+        y = int(random.random() > 0.5)
+        z = np.random.randn(32)
+        data.append(
+            Document(
+                {
+                    "id": id_,
+                    "x": x,
+                    "y": y,
+                    "z": z,
+                }
+            )
+        )
+
+    db.add(schema)
+
+    db.execute(
+        Collection(collection_name).insert_many(data, schema=schema.identifier),
+    )
```

Comment:
This is not the same as ibis, because mongodb supports multiple schemas in a collection, so the schema is associated with the data.

Reply:
Got it.